exchanges/gateio/gateio.go

```diff
@@ -421,169 +933,2821 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	startingTime, err := time.Parse("2006-Jan-02", "2020-Apr-10")
+	if err != nil {
+		return nil, err
+	}
+	if err := common.StartEndTimeCheck(startingTime, from); err == nil {
+		params.Set("from", strconv.FormatInt(from.Unix(), 10))
+	}
+	if err := common.StartEndTimeCheck(from, to); err == nil {
+		params.Set("to", strconv.FormatInt(to.Unix(), 10))
+	}
+	if offset > 0 {
+		params.Set("offset", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var response []SubAccountTransferResponse
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot,
+		http.MethodGet, walletSubAccountTransfer, params, nil, &response)
+}
+
+// GetWithdrawalStatus retrieves withdrawal status
+func (g *Gateio) GetWithdrawalStatus(ctx context.Context, ccy currency.Code) ([]WithdrawalStatus, error) {
+	params := url.Values{}
+	if !ccy.IsEmpty() {
+		params.Set("currency", ccy.String())
+	}
+	var response []WithdrawalStatus
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletWithdrawStatus, params, nil, &response)
+}
+
+// GetSubAccountBalances retrieve sub account balances
+func (g *Gateio) GetSubAccountBalances(ctx context.Context, subAccountUserID string) ([]SubAccountBalance, error) {
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	var response []SubAccountBalance
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletSubAccountBalance, params, nil, &response)
+}
+
+// GetSubAccountMarginBalances query sub accounts' margin balances
+func (g *Gateio) GetSubAccountMarginBalances(ctx context.Context, subAccountUserID string) ([]SubAccountMarginBalance, error) {
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	var response []SubAccountMarginBalance
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletSubAccountMarginBalance, params, nil, &response)
+}
+
+// GetSubAccountFuturesBalances retrieves sub accounts' futures account balances
+func (g *Gateio) GetSubAccountFuturesBalances(ctx context.Context, subAccountUserID, settle string) ([]SubAccountBalance, error) {
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	if settle != "" {
+		params.Set("settle", settle)
+	}
+	var response []SubAccountBalance
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletSubAccountFuturesBalance, params, nil, &response)
+}
+
+// GetSubAccountCrossMarginBalances query subaccount's cross_margin account info
+func (g *Gateio) GetSubAccountCrossMarginBalances(ctx context.Context, subAccountUserID string) ([]SubAccountCrossMarginInfo, error) {
```

Comment:
CrossMarginBalance missing: https://www.gate.io/docs/developers/apiv4/en/#query-subaccount-s-cross-margin-account-info
It would appear you're converting json example data into structs, but you cannot rely upon example data always containing all fields.